### PR TITLE
Add troubleshooting section to README.md

### PR DESCRIPTION
**Pull request by andregasser**
Plus some minor cleanup.

Partially resolves https://github.com/andregasser/bigbone/issues/199.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,16 +131,33 @@ Dependencies:
 </dependency>
 ```
 
-### Usage Examples
+# Using BigBone
+For an easy introduction to using the library, I recommend that you take a look at the sample-* modules of this
```
Comment (bocops) [inline README.md:135]: This should probably be "we" or something neutral instead of "I".

File: README.md [hunk 2]
```diff
@@ -131,16 +131,33 @@ Dependencies:
...
 
-We provide some example code snippets for Java and Kotlin in [USAGE.md](USAGE.md) to get you started.
-
-### Sample Code
-
-There are also two Gradle modules that provide some insights on how to use this library:
 - `sample-java` for Java example code
 - `sample-kotlin` for Kotlin example code
 
+You can run them on your machine. Usually, they required some program arguments to be set before launch. Please
```
Comment (bocops) [inline README.md:141]: Should be "require", not "required".

File: README.md [hunk 3]
```diff
@@ -131,16 +131,33 @@ Dependencies:
...
+
+We also provide some guidance in [USAGE.md](USAGE.md).
+
+# Troubleshooting
+In this section we describe problems that have been reported to us by users of the library. Please 
+read this section before you start working with the library, as the tips given in this section might 
+save you some time.
+
+## Request Not Sent / No Data Returned
+Make sure, you call `.execute()` on the BigBone library method you want to use. Here's an example:
```
Comment (bocops) [inline README.md:152]: Remove comma after "sure"

**Review by bocops — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (andregasser) [reply README.md:135]: Good point. Was not sure about whether to use I or we. But I think it is better to use we in the long term. Will change it and update this PR.

Comment (andregasser) [reply README.md]: Updated.

File: README.md (hunk 2 above)
Comment (andregasser) [reply README.md]: Thanks, fixed!

File: README.md (hunk 3 above)
Comment (andregasser) [reply README.md]: Thanks, fixed!

**Pull request merged**